transport/ProbeServer.cpp

```diff
@@ -0,0 +1,224 @@
+#include "transport/ProbeServer.h"
+#include "logger/Logger.h"
+#include "transport/ice/Stun.h"
+#include "utils/Time.h"
+#include <unistd.h>
+
+namespace transport
+{
+ProbeServer::ProbeServer(ice::IceConfig& iceConfig) : _iceConfig(iceConfig)
+{
+    // TODO copied from IceSession.cpp; refactor
+    char ufrag[14 + 1];
+    char pwd[24 + 1]; // length selected to make attribute *4 length
+
+    generateCredentialString(ufrag, sizeof(ufrag) - 1);
+    generateCredentialString(pwd, sizeof(pwd) - 1);
+
+    _probeCredentials = std::make_pair<std::string, std::string>(ufrag, pwd);
+}
+
+// Endpoint::IEvents
+void ProbeServer::onRtpReceived(Endpoint& endpoint,
+    const SocketAddress& source,
+    const SocketAddress& target,
+    memory::UniquePacket packet)
+{
+}
+
+void ProbeServer::onDtlsReceived(Endpoint& endpoint,
+    const SocketAddress& source,
+    const SocketAddress& target,
+    memory::UniquePacket packet){};
+
+void ProbeServer::onRtcpReceived(Endpoint& endpoint,
+    const SocketAddress& source,
+    const SocketAddress& target,
+    memory::UniquePacket packet)
+{
+}
+
+void ProbeServer::onIceReceived(Endpoint& endpoint,
+    const SocketAddress& source,
+    const SocketAddress& target,
+    memory::UniquePacket packet)
+{
+    replyStunOk(endpoint, source, std::move(packet));
+}
+
+void ProbeServer::onRegistered(Endpoint& endpoint)
+{
+    const std::lock_guard<std::mutex> locker(_endpointsLock);
```

Comment:
I guess we can have it here since it is calle donce at startup, but this is one of the RT workers and they may not pass a mutex at any time.

Reply:
I can remove all the mutexes from ProbeServer.

Reply:
Removed the mutexes.